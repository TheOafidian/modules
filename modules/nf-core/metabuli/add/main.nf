process METABULI_ADD {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/metabuli:1.0.5--pl5321h6a68c12_1':
        'biocontainers/metabuli:1.0.5--pl5321h6a68c12_1' }"

    input:
    tuple val(meta), path(fasta)
    path taxonomy_names, stageAs: 'taxonomy/names.dmp'
    path taxonomy_nodes, stageAs: 'taxonomy/nodes.dmp'
    path taxonomy_merged, stageAs: 'taxonomy/merged.dmp'
    path accession2taxid, stageAs: 'taxonomy/*'

    output:
    tuple val(meta), path("$prefix"), emit: db
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    make_merged = taxonomy_merged ? "" : "touch ${prefix}/taxonomy/merged.dmp"

    """
    mkdir -p $prefix
    mv "taxonomy" $prefix
    $make_merged

    realpath $fasta > fasta.txt
    metabuli add-to-library \\
        fasta.txt \\
        $prefix/$accession2taxid \\
        $prefix \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        metabuli: \$(metabuli version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir -p $prefix

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        metabuli: \$(metabuli version)
    END_VERSIONS
    """
}
