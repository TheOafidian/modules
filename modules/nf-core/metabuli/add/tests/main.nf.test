nextflow_process {

    name "Test Process METABULI_ADD"
    script "../main.nf"
    process "METABULI_ADD"

    tag "modules"
    tag "modules_nfcore"
    tag "metabuli"
    tag "metabuli/add"

    test("sarscov2 protein_db") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                        [
                        file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome.fasta', checkIfExists: true),
                        file(params.modules_testdata_base_path +'genomics/sarscov2/genome/proteome.fasta', checkIfExists: true),
                        ]
                    ]
                input[1] = file(params.modules_testdata_base_path +'genomics/sarscov2/metagenome/prot_names.dmp', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path +'genomics/sarscov2/metagenome/prot_nodes.dmp', checkIfExists: true)
                input[3] = [] // needs a merged.dmp
                input[4] = file(params.modules_testdata_base_path + 'delete_me/metabuli/acc2taxid', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot (process.out).match()},
            )
        }
    }

    test("sarscov2 protein_db stub") {

            options "-stub"

            when {
                process {
                    """
                    input[0] = [
                        [ id:'test' ],
                            [
                            file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
                            file(params.test_data['sarscov2']['genome']['proteome_fasta'], checkIfExists: true)
                            ]
                        ]
                    input[1] = file(params.test_data['sarscov2']['metagenome']['prot_names_dmp'], checkIfExists: true)
                    input[2] = file(params.test_data['sarscov2']['metagenome']['prot_nodes_dmp'], checkIfExists: true)
                    input[3] = [] // needs a merged.dmp
                    input[4] = file(params.modules_testdata_base_path + 'delete_me/metabuli/acc2taxid', checkIfExists: true)
                    """
                }
            }

            then {
                assertAll(
                    { assert process.success },
                    { assert snapshot (process.out).match()},
                )
            }
        }
}
