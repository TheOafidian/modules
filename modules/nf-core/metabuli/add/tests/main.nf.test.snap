{
    "sarscov2 protein_db": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "694009.fna:md5,e73599798195a519ba2565c3f0275b93"
                            ],
                            [
                                "acc2taxid:md5,4ae5ac3b865cb7bf301c3cc72153a7a8",
                                "merged.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "names.dmp:md5,130f9132095562e09c732679c562f5e9",
                                "nodes.dmp:md5,c471c27a4ce85ae74d2c63633c9ce1e3"
                            ],
                            "unmapped.txt:md5,b8c64bc37b7dc465fdaf61b3cd90c29a"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,2a8623d8302b84cba425ddf9381464d4"
                ],
                "db": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "694009.fna:md5,e73599798195a519ba2565c3f0275b93"
                            ],
                            [
                                "acc2taxid:md5,4ae5ac3b865cb7bf301c3cc72153a7a8",
                                "merged.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "names.dmp:md5,130f9132095562e09c732679c562f5e9",
                                "nodes.dmp:md5,c471c27a4ce85ae74d2c63633c9ce1e3"
                            ],
                            "unmapped.txt:md5,b8c64bc37b7dc465fdaf61b3cd90c29a"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2a8623d8302b84cba425ddf9381464d4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T12:01:48.840753727"
    },
    "sarscov2 protein_db stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [

                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,2a8623d8302b84cba425ddf9381464d4"
                ],
                "db": [
                    [
                        {
                            "id": "test"
                        },
                        [

                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2a8623d8302b84cba425ddf9381464d4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T11:40:58.903080427"
    }
}
